core/src/main/java/org/apache/gravitino/stats/storage/LancePartitionStatisticStorage.java

```diff
@@ -0,0 +1,427 @@
+
+      String filter = "table_id = " + tableId + partitionFilter;
+
+      try (LanceScanner scanner =
+          dataset.newScan(
+              new ScanOptions.Builder()
+                  .columns(
+                      Arrays.asList(
+                          TABLE_ID_COLUMN,
+                          PARTITION_NAME_COLUMN,
+                          STATISTIC_NAME_COLUMN,
+                          STATISTIC_VALUE_COLUMN,
+                          AUDIT_INFO_COLUMN))
+                  .withRowId(true)
+                  .batchSize(readBatchSize)
+                  .filter(filter)
+                  .build())) {
+        Map<String, List<PersistedStatistic>> partitionStatistics = Maps.newConcurrentMap();
+        try (ArrowReader reader = scanner.scanBatches()) {
+          while (reader.loadNextBatch()) {
+            VectorSchemaRoot root = reader.getVectorSchemaRoot();
+            List<FieldVector> fieldVectors = root.getFieldVectors();
+            VarCharVector partitionNameVector = (VarCharVector) fieldVectors.get(1);
+            VarCharVector statisticNameVector = (VarCharVector) fieldVectors.get(2);
+            LargeVarCharVector statisticValueVector = (LargeVarCharVector) fieldVectors.get(3);
+            VarCharVector auditInfoNameVector = (VarCharVector) fieldVectors.get(4);
+
+            for (int i = 0; i < root.getRowCount(); i++) {
+              String partitionName = new String(partitionNameVector.get(i), StandardCharsets.UTF_8);
+              String statisticName = new String(statisticNameVector.get(i), StandardCharsets.UTF_8);
+              String statisticValueStr =
+                  new String(statisticValueVector.get(i), StandardCharsets.UTF_8);
+              String auditInoStr = new String(auditInfoNameVector.get(i), StandardCharsets.UTF_8);
+
+              StatisticValue<?> statisticValue =
+                  JsonUtils.anyFieldMapper().readValue(statisticValueStr, StatisticValue.class);
+              AuditInfo auditInfo =
+                  JsonUtils.anyFieldMapper().readValue(auditInoStr, AuditInfo.class);
+
+              PersistedStatistic persistedStatistic =
+                  PersistedStatistic.of(statisticName, statisticValue, auditInfo);
+
+              partitionStatistics
+                  .computeIfAbsent(partitionName, k -> Lists.newArrayList())
+                  .add(persistedStatistic);
+            }
+          }
+
+          return partitionStatistics.entrySet().stream()
+              .map(entry -> PersistedPartitionStatistics.of(entry.getKey(), entry.getValue()))
+              .collect(Collectors.toList());
+        }
+      } catch (Exception e) {
+        throw new RuntimeException(e);
+      }
+    }
+  }
+
+  private Dataset open(String fileName) {
+    try {
+      return Dataset.open(fileName, allocator);
+    } catch (IllegalArgumentException illegalArgumentException) {
+      if (illegalArgumentException.getMessage().contains("was not found")) {
```

Comment:
Do we have better way to check if the lance file exists or not? Using exception message is too tricky.